app/models/conversation.rb

```diff
@@ -0,0 +1,28 @@
+class Conversation < ApplicationRecord
+  belongs_to(
+    :author,
+    class_name: User.name,
+    foreign_key: "author_user_id",
+  )
+  belongs_to(
+    :recipient,
+    class_name: User.name,
+    foreign_key: "recipient_user_id",
+  )
+  has_many :messages
+
+  validates :short_id, presence: true, uniqueness: true
+  validates :subject, presence: true
+
+  scope :involving, ->(user) do
```

Comment:
Nice scope. I'm glad to see fewer and fewer direct `where` calls outside of models.